backends/firrtl/firrtl.cc

```diff
@@ -42,6 +42,77 @@ static const FDirection FD_OUT = 0x2;
 static const FDirection FD_INOUT = 0x3;
 static const int FIRRTL_MAX_DSH_WIDTH_ERROR = 20; // For historic reasons, this is actually one greater than the maximum allowed shift width
 
+// Shamelessly copied from ilang_backend.cc. Something better is surely possible here.
```

Comment:
If this is character for character identical then why not just call that function `ILANG_BACKEND::dump_const()`?

Reply:
The various yosys backends seemed independent to me so I didn't want to introduce a dependency between the firrtl and ilang backend (though I admit I didn't look very deeply). But I suppose the ilang backend has little chance of changing and I could just call its `dump_const()` function (it is indeed a character for character copy).

Reply:
I would say the `ILANG_BACKEND::*` are okay to call from outside because ilang is probably the most native format (and other files like `kernel/log.cc` and `passes/cmds/bugpoint.cc` also call `ILANG_BACKEND::*` functions)